[BUG] config cache not supported

**Describe the bug**
 Task `:nexusIQScan` of type `org.sonatype.gradle.plugins.scan.nexus.iq.scan.NexusIqScanTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/8.1.1/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution


**To Reproduce**
`./gradlew nexusIQScan`

**Expected behavior**
Works

**Screenshots**

**Desktop (please complete the following information):**
 - OS: OS X 13.3.1
 - Gradle Version: 8.1.1
 - JVM Version and Flavor: Java 17, Open JDK
 - Nexus IQ Server Version: 153

**Additional context**
workaround is `org.gradle.unsafe.configuration-cache=false`

Trying with https://github.com/Morfly/MultiModuleMovies as it apparently replicates the issue, although I see the application evaluation is done and the report is on the output.

The error "Invocation of 'Task.project' by task ':nexusIQScan' at execution time is unsupported." is show below the report:

```
Policy Action: None
Number of components affected: 113 critical, 0 severe, 0 moderate
Number of grandfathered policy violations: 215
Number of components: 209
The detailed report can be viewed online at http://localhost:8070/ui/links/application/movies-compose-multi-module/report/beb4ab9e18554f04a2e5aecd82b477ed


FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.

2 problems were found storing the configuration cache, 1 of which seems unique.
- Task `:nexusIQScan` of type `org.sonatype.gradle.plugins.scan.nexus.iq.scan.NexusIqScanTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution

See the complete report at file:///folder/MultiModuleMovies/build/reports/configuration-cache/79i2xjc9862gsyhmov9lxncs8/cu8tk4l7hk0g2ef5h9cnoamsw/configuration-cache-report.html
> Invocation of 'Task.project' by task ':nexusIQScan' at execution time is unsupported.
```

Granted, even though the result is "Policy Action: None", the plugin exits with "FAILURE" due to this issue.

There are 2 stacktraces in the mentioned HTML file:

```
org.gradle.api.InvalidUserCodeException: Invocation of 'Task.project' by task ':nexusIQScan' at execution time is unsupported.
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onTaskExecutionAccessProblem(ConfigurationCacheProblemsListener.kt:74)
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onProjectAccess(ConfigurationCacheProblemsListener.kt:53)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:464)
	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:446)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:61)
	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast$ListenerDispatch.dispatch(DefaultListenerManager.java:434)
	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast.dispatch(DefaultListenerManager.java:221)
	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast.dispatch(DefaultListenerManager.java:192)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy91.onProjectAccess(Unknown Source)
	at org.gradle.api.internal.AbstractTask.notifyProjectAccess(AbstractTask.java:1053)
	at org.gradle.api.internal.AbstractTask.getProject(AbstractTask.java:237)
	at org.gradle.api.DefaultTask.getProject(DefaultTask.java:59)
	at org.sonatype.gradle.plugins.scan.nexus.iq.scan.NexusIqScanTask.buildUserAgent(NexusIqScanTask.java:213)
	at org.sonatype.gradle.plugins.scan.nexus.iq.scan.NexusIqScanTask.scan(NexusIqScanTask.java:96)
```

```
org.gradle.api.InvalidUserCodeException: Invocation of 'Task.project' by task ':nexusIQScan' at execution time is unsupported.
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onTaskExecutionAccessProblem(ConfigurationCacheProblemsListener.kt:74)
	at org.gradle.configurationcache.initialization.DefaultConfigurationCacheProblemsListener.onProjectAccess(ConfigurationCacheProblemsListener.kt:53)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:464)
	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:446)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:61)
	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast$ListenerDispatch.dispatch(DefaultListenerManager.java:434)
	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast.dispatch(DefaultListenerManager.java:221)
	at org.gradle.internal.event.DefaultListenerManager$EventBroadcast.dispatch(DefaultListenerManager.java:192)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy91.onProjectAccess(Unknown Source)
	at org.gradle.api.internal.AbstractTask.notifyProjectAccess(AbstractTask.java:1053)
	at org.gradle.api.internal.AbstractTask.getProject(AbstractTask.java:237)
	at org.gradle.api.DefaultTask.getProject(DefaultTask.java:59)
	at org.sonatype.gradle.plugins.scan.nexus.iq.scan.NexusIqScanTask.scan(NexusIqScanTask.java:107)
```

--- a/src/main/java/org/sonatype/gradle/plugins/scan/ScanPlugin.java
+++ b/src/main/java/org/sonatype/gradle/plugins/scan/ScanPlugin.java
@@ -30,15 +30,27 @@ import org.gradle.api.Project;
 
 public class ScanPlugin implements Plugin<Project>
 {
-  private static final boolean IS_GRADLE_MIN_49 = GradleVersion.current().compareTo(GradleVersion.version("4.9-rc-1"))>=0;
+  private static final boolean IS_GRADLE_MIN_4_9 =
+      GradleVersion.current().compareTo(GradleVersion.version("4.9-rc-1")) >= 0;
+
+  private static final boolean IS_GRADLE_MIN_7_4 =
+      GradleVersion.current().compareTo(GradleVersion.version("7.4")) >= 0;
+
   private static final String SONATYPE_GROUP = "Sonatype";
 
+  private static final String TASK_NOT_COMPATIBLE_WITH_CONFIG_CACHE_REASON =
+      "Task needs to access the project configuration";
+
   @Override
+  @SuppressWarnings("deprecation")
   public void apply(Project project) {
     project.getExtensions().create("nexusIQScan", NexusIqPluginScanExtension.class, project);
     createTask(project, "nexusIQScan", NexusIqScanTask.class, task -> {
       task.setGroup(SONATYPE_GROUP);
       task.setDescription("Scan and evaluate the dependencies of the project using Nexus IQ Server.");
+      if (IS_GRADLE_MIN_7_4) {
+        task.notCompatibleWithConfigurationCache(TASK_NOT_COMPATIBLE_WITH_CONFIG_CACHE_REASON);
+      }
     });
 
     project.getExtensions().create("nexusIQIndex", NexusIqPluginIndexExtension.class, project);
@@ -46,17 +58,23 @@ public class ScanPlugin implements Plugin<Project>
       task.setGroup(SONATYPE_GROUP);
       task.setDescription("Saves information about the dependencies of a project into module information "
             + "(module.xml) files that Sonatype CI tools can use to include these dependencies in a scan.");
+      if (IS_GRADLE_MIN_7_4) {
+        task.notCompatibleWithConfigurationCache(TASK_NOT_COMPATIBLE_WITH_CONFIG_CACHE_REASON);
+      }
     });
 
     project.getExtensions().create("ossIndexAudit", OssIndexPluginExtension.class, project);
     createTask(project, "ossIndexAudit", OssIndexAuditTask.class, task -> {
       task.setGroup(SONATYPE_GROUP);
       task.setDescription("Audit the dependencies of the project using OSS Index.");
+      if (IS_GRADLE_MIN_7_4) {
+        task.notCompatibleWithConfigurationCache(TASK_NOT_COMPATIBLE_WITH_CONFIG_CACHE_REASON);
+      }
     });
   }
 
   private static <T extends Task> void createTask(Project project, String name, Class<T> type, Action<? super T> configuration) {
-    if (IS_GRADLE_MIN_49) {
+    if (IS_GRADLE_MIN_4_9) {
       project.getTasks().register(name, type, configuration);
     } else {
       project.getTasks().create(name, type, configuration);

